src/cluster_legacy.c

```diff
@@ -2149,6 +2166,18 @@ void clusterProcessGossipSection(clusterMsg *hdr, clusterLink *link) {
                 !(flags & CLUSTER_NODE_NOADDR) &&
                 !clusterBlacklistExists(g->nodename))
             {
+                /* Prevent a node with an invalid ID from being added to the nodes dict.
+                 * An invalid ID indicates memory corruption on the sender side. */
+                if (verifyClusterNodeId(g->nodename, CLUSTER_NAMELEN) != C_OK) {
```

Comment:
As discussed, let's pull this to the top so we ignore ALL of the gossip section if we detect any corruption.